pkg/api/steve/proxy/proxy.go

```diff
@@ -87,8 +87,10 @@ func routeToShellProxy(key, value string, localSupport bool, localCluster http.H
 			if localSupport {
 				authed := proxyHandler.userCanAccessCluster(r, cluster)
 				if !authed {
-					rw.WriteHeader(http.StatusUnauthorized)
-					return
+					if r.Context().Value(auth.CattleAuthFailed) == nil {
```

Comment:
nit here and below: 
```suggestion
					if r.Context().Value(auth.CattleAuthFailed) != "true" {
```

Reply:
I agree here that we could be more explicit.